src/main/java/com/minecolonies/coremod/colony/buildings/BuildingCowboy.java

```diff
@@ -67,11 +76,48 @@ public AbstractJob createJob(final CitizenData citizen)
         return new JobCowboy(citizen);
     }
 
+    @Override
+    public void serializeToView(@NotNull final ByteBuf buf)
+    {
+        super.serializeToView(buf);
+        buf.writeBoolean(milkCows);
+    }
+
+    @Override
+    public void writeToNBT(@NotNull final NBTTagCompound compound)
+    {
+        super.writeToNBT(compound);
+        compound.setBoolean(NBT_MILK_COWS, this.milkCows);
+    }
+
+    @Override
+    public void readFromNBT(@NotNull final NBTTagCompound compound)
+    {
+        super.readFromNBT(compound);
+        this.milkCows = compound.getBoolean(NBT_MILK_COWS);
+    }
+
+    public boolean isMilkCows()
+    {
+        return milkCows;
+    }
+
+    public void setMilkCows(final boolean milkCows)
+    {
+        this.milkCows = milkCows;
+        markDirty();
+    }
+
     /**
      * ClientSide representation of the building.
      */
     public static class View extends AbstractBuildingWorker.View
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce this class from 41 to the maximum allowed 30 or externalize it in a public class. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)